File: pmid_record.ts
import { verifyIdentifier } from "./utility.ts";
import { normalizePMID, validatePMID } from "./pmid.ts";

/**
 * Verifies an PMID using an external database, pmid.org
 * @param pmid - The identifier string to verify.
 * @returns Promise<boolean> - True if the PMID exists, otherwise false.
 */
export async function verifyPMID(pmid: string): Promise<boolean> {
  const normalizedID = normalizePMID(pmid);
  return await verifyIdentifier(
    pmid,
    `https://pubmed.ncbi.nlm.nih.gov/${normalizedID}/`,
    validatePMID,
  );
}

File: pmid_record.go
package metadatatools

// VerifyPMID verifies a PMID exists via pubmed.ncbi.nlm.nih.gov. The
// identifier is normalized before verifying.
func VerifyPMID(pmid string) bool {
	normalized := NormalizePMID(pmid)
	return VerifyIdentifier(pmid, "https://pubmed.ncbi.nlm.nih.gov/"+normalized+"/", ValidatePMID)
}
